### PR TITLE
fix(kb): give knowledge bubbles their own sparse-checkout defaults

### PR DESCRIPTION
**Pull request by galexy**
## What broke

Knowledge Bubbles were syncing "successfully" while their knowledge files never landed on disk.

Observed on the product bubble (`kb_019f18ee-2eb0-7386-a1d6-d5176d40222c`) at commit `6175710`:

- **In the commit:** 9 files under `knowledge/`
- **On disk:** 1 (`knowledge/agents.md`)
- **The other 8:** present in the object store, carrying the skip-worktree bit

No data loss — the files were always in git. They were never written into the working tree. An AI coworker reading that bubble saw a near-empty knowledge base with no signal that anything was missing.

This affected **every bubble on every machine**, and repeated on **every sync pass**, so it was not self-correcting.

## Root cause

Each synced repo carries `.sageox/sync.manifest` declaring which paths to materialize. The server seeds new bubbles with a comment-only file:

```
# managed by SageOx KB watchman
```

No `version` directive, so `manifest.Parse` rejects it and `ParseFile` falls back to a hardcoded include set. **The only fallback set that existed was the team-context one** — `SOUL.md`, `docs/`, `discussions/`, `coworkers/`. No `knowledge/`.

Both the clone path and every pull recompute sparse from that same unparseable manifest, so the exclusion re-asserted continuously. A manual `git sparse-checkout` widening would not have survived the next daemon pass.

```mermaid
flowchart TD
    A["Server seeds bubble with<br/>comment-only sync.manifest"] --> B["manifest.Parse → ErrNoVersion"]
    B --> C["ParseFile falls back"]
    C --> D{"Which fallback set?"}
    D -->|"before: only one existed"| E["team-context includes<br/>(no knowledge/)"]
    E --> F["knowledge/ excluded from sparse set"]
    F --> G["TwoPhaseClone applies it"]
    F --> H["every pull re-applies it"]
    G --> I["knowledge/ invisible on disk"]
    H --> I
    D -->|"after this PR"| J["kb includes<br/>.sageox/, AGENTS.md, knowledge/"]
    J --> K["knowledge/ materializes"]
```

### Why it looked partial instead of total

`--no-cone` sparse-checkout uses gitignore matching, where a pattern with no interior slash matches at **any depth**. The team-context set's bare `AGENTS.md` entry (intended for the repo-root file) also matched `knowledge/agents.md` — case-insensitively, since `core.ignorecase=true` on macOS.

That leaked exactly one file out of an otherwise fully-excluded directory, making a structural failure present as a file-specific one. Real triage cost.

## What this PR ships

| Change | File |
|---|---|
| `RepoKind` type + kb-specific fallback set (`.sageox/`, `AGENTS.md`, `knowledge/`) | `internal/manifest/fallback.go` |
| `ParseFile` and `TwoPhaseClone` now **require** a repo kind | `internal/manifest/parser.go`, `internal/gitserver/two_phase_clone.go` |
| `ComputeSparseSet` root-anchors every include (`AGENTS.md` → `/AGENTS.md`) | `internal/manifest/parser.go` |
| Shared `EnsureSageoxInclude` replaces two hand-rolled copies | `internal/gitserver/two_phase_clone.go`, `cmd/ox/doctor_team.go` |
| Call sites declare their kind (kb vs team-context) | `internal/daemon/sync_bubbles.go`, `internal/daemon/sync_team.go`, `cmd/ox/doctor_git_repos_fix.go` |

**Anti-recurrence:** there is deliberately no default repo kind. Previously "unspecified" silently meant team-context, which is the trapdoor this bug fell through. Omitting it is now a compile error.

**Anchoring is a no-op for team contexts.** Patterns with an interior slash are already root-relative under gitignore rules, and anything under an included directory (`coworkers/`, `docs/`) stays included regardless.

### Latent bug fixed in passing

The two copies of the "ensure `.sageox/` is in the set" guard were not identical. `two_phase_clone.go` appended; `doctor_team.go` **prepended**:

```go
sparsePaths = append([]string{".sageox/"}, sparsePaths...)
```

Order is load-bearing — later patterns override earlier ones, and `!/*/` excludes all root directories. Prepending puts `.sageox/` ahead of `!/*/`, which then re-excludes it. Verified empirically: the prepended form leaves `.sageox` absent from the working tree.

So `ox doctor`'s repair for a broken sparse checkout could itself hide the manifest — the exact state `doctor_kb_repo_health.go` warns about elsewhere. Low real-world impact (the branch only fires when `.sageox/` is already missing from the computed set, which real manifests don't hit), but it lived on the recovery path. The shared helper appends, with a test asserting the ordering.

## Test Plan

**Both fixes verified load-bearing** — each was reverted individually and the tests confirmed red, then restored:

- Neuter the kb fallback → `TestSyncBubbles_UnparseableManifest_MaterializesKnowledgeTree` fails on clone **and** pull, and `discussions/` leaks into the bubble.
- Neuter anchoring → `TestSparseCheckout_BareFilePatternDoesNotLeakNestedMatches` shows `knowledge/agents.md` and `knowledge/AGENTS.md` materializing from a manifest that only asked for the root file.

**New coverage:**

- `TestSyncBubbles_UnparseableManifest_MaterializesKnowledgeTree` — end-to-end through the daemon with the **verbatim** server-seeded manifest, real git, asserting bubble shape after clone *and* after pull (sparse is recomputed every pull, so a wrong fallback re-hides the tree even if clone got it right). Asserts both directions: `knowledge/` appears, `discussions/` does not.
- `TestSparseCheckout_BareFilePatternDoesNotLeakNestedMatches` — drives real git rather than asserting on pattern strings. Covers the exact-case nested match, the case-variant nested match, and confirms the intended root file still lands.
- `TestFallbackConfigFor_KB` / `_TeamContext` — the two sets cannot drift into each other.
- `TestComputeSparseSet_AnchorsPatternsToRoot`, `TestEnsureSageoxInclude`.

**Against real production data:** applying the corrected sparse set to a copy of `kb_019f18ee-...` takes `knowledge/` from 1 file to all 9, with zero skip-worktree entries remaining.

**Gates:** `make lint` clean (0 issues). `internal/manifest`, `internal/gitserver`, `internal/daemon`, `internal/daemon/agentwork`, `internal/daemon/hooks`, `internal/daemon/testutil` all pass.

**Pre-existing failures, not from this change** (verified by re-running with all changes stashed): `TestCodeDBMaintainerNilManager`, `TestDoctorFreshInstall_NoWarnings`, `TestDoctorFreshInstall_EmptyRepo_NoWarnings`. The last two read the developer's real `~/.local/share/sageox` — filed as `ox-mael`.

## Rollout

Existing broken bubbles heal on the next daemon pull once running a build with this fix — sparse is reapplied from the manifest on every pass. No reclone needed.

## Follow-ups filed

| Issue | P | What |
|---|---|---|
| `ox-8kly` | P2 | Server still seeds a content-free manifest. Harmless now, but the server can't change a bubble's sparse shape without an ox release. |
| `ox-mael` | P2 | `TestDoctorFreshInstall_*` reads real machine state instead of a sandbox. |
| `ox-znck` | P3 | `ox doctor` checked only for `.sageox` in the cone, so it reported these bubbles healthy the entire time. It should flag a bubble whose `knowledge/` is missing. |

Closes `ox-182q`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

SageOx-Session: https://sageox.ai/c/ses_019fb602-3c60-7f17-a7e9-ab8c6073b8a2

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved sparse checkout for team-context repositories and knowledge bubbles.
  * Preserved required `.sageox/`, `knowledge/`, and `AGENTS.md` content when manifests are missing or unreadable.
  * Prevented similarly named files in nested directories from being included unintentionally.
  * Standardized repository-specific fallback handling during cloning and synchronization.
* **Tests**
  * Added coverage for fallback manifests, sparse path matching, and synchronization after cloning or pulling.

### CODE REVIEW

File: internal/gitserver/two_phase_clone.go [hunk 1]
```diff
@@ -55,7 +60,7 @@ var TestAllowFileTransport bool
 //     1. SparseCheckoutDirectories uses --no-cone mode (ox uses file-level patterns)
 //     2. go-git network performance vs native git for clone
 //     3. credential injection (URL-embedded tokens) works correctly
-func TwoPhaseClone(ctx context.Context, cloneURL, repoPath string) (*TwoPhaseCloneResult, error) {
+func TwoPhaseClone(ctx context.Context, cloneURL, repoPath string, kind manifest.RepoKind) (*TwoPhaseCloneResult, error) {
```
[the remark this answers is not in the document]
Comment (galexy) [reply internal/gitserver/two_phase_clone.go:63]: Fixed in 56ed92ec. All four calls in `internal/daemon/twin_clone_test.go` (lines 69, 112, 144, 170) now pass `manifest.RepoKindTeamContext` — correct for these fixtures, which seed `SOUL.md` and are named `TestTwoPhaseClone_BasicTeamContext` etc.

Good catch, and my miss: `go vet ./...` and `make lint` both pass because neither builds `slow`-tagged files. Verified the failure and the fix directly with `go vet -tags=slow ./...`, which is now clean.

**Comment by galexy** [comment]
All four review findings addressed in 56ed92ec.

## P1 — slow-tag build broken (@greptile-apps, @coderabbitai)

**Valid, fixed.** The four `TwoPhaseClone` calls in `internal/daemon/twin_clone_test.go` now pass `manifest.RepoKindTeamContext`. That's the right kind — the fixtures seed `SOUL.md` and the tests are named `TestTwoPhaseClone_BasicTeamContext` / `_NoManifest_FallsBack` / etc.

My verification gap, worth naming: `go build ./...`, `go vet ./...` and `make lint` all pass while this file is broken, because none of them build `slow`-tagged sources. `make test-preflight` would have caught it; I ran targeted tests instead. Now verified with `go vet -tags=slow ./...` and `go build -tags=slow ./...`, both clean.

## Nitpick — `os.IsNotExist` → `errors.Is` (`internal/manifest/parser.go`)

**Valid, fixed.** Matches the repo rule in `CLAUDE.md` ("Errors: Use `errors.Is()`/`errors.As()`"). `errors` was already imported.

## Nitpick — unknown-kind fallback contradicts its own doc (`internal/manifest/fallback.go`)

**Valid, fixed — and the strongest of the four.** The doc comment claimed an unset kind was "a plumbing bug, not a supported input" while the code logged a warning and returned the team-context set. That is precisely the failure mode this PR exists to remove: a wrong tree, announced only in a log line nobody reads. A future `RepoKindLedger` added without a case here would have reproduced the original bug exactly.

`fallbackIncludesFor` now panics on an unrecognized kind, with `TestFallbackConfigFor_UnknownKindPanics` covering the empty string and two unknown values.

I weighed panicking in daemon-resident code and accepted it on these grounds, recorded in the function comment: it is unreachable from every current call path (all six callers pass one of the two constants, and the compiler requires the argument), so it can only fire in a developer's first test run after adding a kind — which is exactly when you want it. The comment also flags the boundary condition: if a `RepoKind` ever originates from config or an API response, it must be validated there, or this becomes a remotely-triggerable crash.

## Not addressed

**Docstring coverage 56% vs 80% threshold.** Every exported symbol this PR adds is documented. The shortfall is in surrounding pre-existing code, and expanding it would enlarge the diff without serving the fix. Happy to take it as a separate pass if you want the package brought over the line.

---

Gates after these changes: `make lint` 0 issues; `go build`/`go vet` clean under both default and `slow` tags; `internal/manifest`, `internal/gitserver`, and `internal/daemon` all pass.
